spring-cloud-stream/src/main/java/org/springframework/cloud/stream/endpoint/BindingsEndpoint.java

```diff
@@ -62,25 +53,41 @@ public BindingsEndpoint(List<InputBindingLifecycle> inputBindingLifecycles) {
 		this.objectMapper = new ObjectMapper();
 	}
 	
-	@ReadOperation
-	public List<?> bindings() {
-		return this.objectMapper.convertValue(this.gatherInputBindings(), List.class);	
-	}
-	
-	public StopEndpoint getStopEndpoint() {
-		return new StopEndpoint();
-	}
-	
-	public StartEndpoint getStartEndpoint() {
-		return new StartEndpoint();
+	@WriteOperation
+	public boolean changeState(@Selector String name, State state) {
+		Assert.notNull(name, "'name' must not be null");
+		Assert.notNull(name, "'state' must not be null");
```

Comment:
If these asserts ever do anything, there's a bug in Spring Boot. You should only get null if an argument is annotated with `@Nullable`. Your endpoint won't even be called if a required parameter is missing. For example, a `POST` that's missing a `state` will automatically be rejected with a 400 response.

Reply:
Looks like the the second `Assert` must be against `state` param, but `name` again :smile:

Reply:
yeah, I am removing both based on Andy's explanation